### PR TITLE
Feature/major refactoring

### PR DESCRIPTION
**Pull request by mikeslgoh**
- Created BaseWorldManager
- Moved player related functions to PlayerManager
- Renamed scene names as well as updated for SceneData

NOTE: Have not stylecopped yet so please forgive me :')

### CODE REVIEW

File: Assets/Script/Managers/WorldManagers/BaseWorldManager.cs [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+using UnityEngine.Assertions;
+
+
+/// <summary>
+/// Manager for all FixedUpdate methods
+/// all method that will update with time will be passed through this script
+/// </summary>
+[RequireComponent(typeof(PlayerManager))]
+[RequireComponent(typeof(ParkingManager))]
+abstract class BaseWorldManager : MonoBehaviour
```
Comment (m4rtint) [inline Assets/Script/Managers/WorldManagers/BaseWorldManager.cs]: public abstract class?

File: Assets/Script/Managers/WorldManagers/DefaultWorldManager.cs [hunk 2]
```diff
@@ -0,0 +1,8 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+class DefaultWorldManager : BaseWorldManager
```
Comment (m4rtint) [inline Assets/Script/Managers/WorldManagers/DefaultWorldManager.cs]: what modifier should this be?

File: Assets/Script/Managers/WorldManagers/BaseWorldManager.cs (hunk 1 above)
Comment (mikeslgoh) [reply Assets/Script/Managers/WorldManagers/BaseWorldManager.cs]: Ah didn't realise I didn't add that there

File: Assets/Script/Managers/WorldManagers/DefaultWorldManager.cs (hunk 2 above)
Comment (mikeslgoh) [reply Assets/Script/Managers/WorldManagers/DefaultWorldManager.cs]: I was getting an error when I had the modifier. Might have been related to the previous comment

**Comment by mikeslgoh** [comment]
[Refactor world manager](https://trello.com/c/QEelCrQq/60-refactor-world-manager)

File: Assets/Script/Managers/WorldManagers/BaseWorldManager.cs [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+public abstract class BaseWorldManager : MonoBehaviour
+{
+    [Header("Managers")]
+    [SerializeField] private PlayerManager m_playerManager;
+    [SerializeField] private ParkingManager m_parkingManager;
+    private NPCManager m_npcManager;
+
+    [Header("Properties")]
+    [Range(1, 4)]
+    [SerializeField] private int m_numberOfPlayers = 1;
```
Comment (m4rtint) [inline Assets/Script/Managers/WorldManagers/BaseWorldManager.cs]: Would it be better if you only have numberofplayers in the PlayerManager.

File: Assets/Script/Managers/PlayerManager.cs [hunk 4]
```diff
@@ -1,122 +1,47 @@
...
-public class WorldManager : MonoBehaviour
+[RequireComponent(typeof(ScoreManager))]
+public class PlayerManager : MonoBehaviour
 {
     private readonly string Player = "Player_";
     private readonly string Camera = "Camera_";
     private Transform[] m_carSpawnPoints;
 
     [Header("Managers")]
     [SerializeField] private ScoreManager m_scoreManager;
```
Comment (m4rtint) [inline Assets/Script/Managers/PlayerManager.cs]: Is it better to put this in the WorldManager?

Comment (mikeslgoh) [reply Assets/Script/Managers/PlayerManager.cs]: I'll look into moving ScoreManager

File: Assets/Script/Managers/PlayerManager.cs [hunk 5]
```diff
@@ -1,32 +1,18 @@
...
-
-/// <summary>
-/// Manager for all FixedUpdate methods
-/// all method that will update with time will be passed through this script
-/// </summary>
-[RequireComponent(typeof(ScoreManager))]
-[RequireComponent(typeof(TargetManager))]
 [RequireComponent(typeof(SplitScreenManager))]
-public class WorldManager : MonoBehaviour
+[RequireComponent(typeof(ScoreManager))]
```
Comment (m4rtint) [inline Assets/Script/Managers/PlayerManager.cs]: Remove this

File: Assets/Script/Managers/WorldManagers/BaseWorldManager.cs [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+        CheckNumberOfPlayers();
+        m_carsInPlay = m_playerManager.Initialize();
+    }
+
+    private void SetNewParkingSpot()
+    {
+        m_playerManager.SetNewParkingSpot(m_parkingManager);
+    }
+
+    private void CheckNumberOfPlayers()
```
Comment (m4rtint) [inline Assets/Script/Managers/WorldManagers/BaseWorldManager.cs]: This should also be moved over to PlayersManager

File: Assets/Script/Managers/WorldManagers/BaseWorldManager.cs [hunk 7]
```diff
@@ -0,0 +1,118 @@
...
+[RequireComponent(typeof(ParkingManager))]
+public abstract class BaseWorldManager : MonoBehaviour
+{
+    [Header("Managers")]
+    [SerializeField] private PlayerManager m_playerManager;
+    [SerializeField] private ParkingManager m_parkingManager;
+    [SerializeField] private ScoreManager m_scoreManager;
+    private NPCManager m_npcManager;
+
+    private List<GameObject> m_carsInPlay = new List<GameObject>();
```
Comment (m4rtint) [inline Assets/Script/Managers/WorldManagers/BaseWorldManager.cs]: should remove this. 
If we need a m_carsInPlay, it should be taken from the PlayerManager

File: Assets/Script/Game/NPCController.cs [hunk 8]
```diff
@@ -61,19 +66,18 @@ private void OnTriggerStay2D(Collider2D collision)
         }
         else
         {
-            //float step = m_rotSpeed * Time.deltaTime;
-            //gameObject.transform.rotation = Quaternion.RotateTowards(transform.rotation, collision.gameObject.transform.rotation, step);
+            ////float step = m_rotSpeed * Time.deltaTime;
```
Comment (mikeslgoh) [inline Assets/Script/Game/NPCController.cs]: why is this commented out twice?

File: Assets/Script/Game/PlayerController.cs [hunk 9]
```diff
@@ -0,0 +1,73 @@
...
+    /// </summary>
+    /// <param name="score">number of points</param>
+    public void UpdateScore(int score)
+    {
+        m_userInterface.UpdateScore(score);
+    }
+
+	private void SetupCar(PlayerControllerInformation playerInfo, Transform spawnPoint)
+    {
+		m_car.tag = TagNames.CarTag;
```
Comment (mikeslgoh) [inline Assets/Script/Game/PlayerController.cs]: the identing here is a little of

**Pull request merged**